app/src/main/java/zapsolutions/zap/customView/WalletSpinner.java

```diff
@@ -0,0 +1,119 @@
+        init();
+    }
+
+    public WalletSpinner(Context context, int mode) {
+        super(context, mode);
+        init();
+    }
+
+    public WalletSpinner(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public WalletSpinner(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+    public WalletSpinner(Context context, AttributeSet attrs, int defStyleAttr, int mode) {
+        super(context, attrs, defStyleAttr, mode);
+        init();
+    }
+
+    public WalletSpinner(Context context, AttributeSet attrs, int defStyleAttr, int mode, Resources.Theme popupTheme) {
+        super(context, attrs, defStyleAttr, mode, popupTheme);
+        init();
+    }
+
+    private OnWalletSpinnerChangedListener mListener;
+
+    private void init() {
+        updateList();
+
+        initFinished = false;
+
+        this.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
+                if (initFinished) {
+                    int lastPos = adapterView.getCount() - 1;
+                    if (position != lastPos) {
+                        String selectedWalletId = WalletConfigsManager.getInstance().getAllWalletConfigs(true).get(position).getId();
+                        if (!WalletConfigsManager.getInstance().getCurrentWalletConfig().getId().equals(selectedWalletId)) {
+                            // Save selected Wallet ID in prefs making it the current wallet.
+                            PrefsUtil.edit().putString(PrefsUtil.CURRENT_WALLET_CONFIG, selectedWalletId).commit();
+
+                            // Update the wallet spinner list, so everything is at it's correct position again.
+                            updateList();
+
+                            // Inform the listener. This is where the new wallet is opened.
+                            mListener.onWalletChanged();
+                        }
+                    } else {
+                        // Open wallet Management
+                        Intent intent = new Intent(getContext(), ManageWalletsActivity.class);
+                        getContext().startActivity(intent);
+
+                        // If going back we don't want to have "Manage.." selected
+                        adapterView.setSelection(0);
+                    }
+                } else {
+                    // When filling the list onItem Selected ist called for the first time.
+                    // In this case we don't want to select something, but mark it ready for interaction instead.
+                    initFinished = true;
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+    }
+
+    private void updateList() {
```

Comment:
We may want to refresh the list also when the `WalletFragment` goes into `onResume`
Reason:
When changing the name of a wallet through spinner->manage wallet, the name in the spinner gets not refreshed.

Reply:
Good catch! I fixed it.